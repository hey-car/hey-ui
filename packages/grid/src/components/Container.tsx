import React from 'react';
import cn from 'classnames';

import { ContainerProps } from './Container.types';

import styles from './Container.module.css';

function Container<E extends React.ElementType = 'div'>({
  className,
  children,
  component,
  ...rest
}: ContainerProps<E>) {
  const Component = component ?? 'div';

  return React.createElement(
    Component,
    {
      className: cn(styles.container, className),
      ...rest,
    },
    children,
  );
}

export default Container;
