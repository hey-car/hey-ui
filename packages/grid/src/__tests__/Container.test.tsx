import React from 'react';
import { render } from '@testing-library/react';

import Container from '../components/Container';

describe('Grid.Container', () => {
  describe('Snapshots tests', () => {
    it('should match snapshot', () => {
      const { container } = render(<Container>content</Container>);

      expect(container).toMatchSnapshot();
    });
  });

  it('should set tag correctly', () => {
    const { getByText } = render(
      <Container component="section">hey</Container>,
    );

    expect(getByText('hey').tagName).toBe('SECTION');
  });
});
