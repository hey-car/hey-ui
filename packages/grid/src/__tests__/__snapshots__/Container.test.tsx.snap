// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid.Container Snapshots tests should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    content
  </div>
</div>
`;
